cache/src/main/java/io/envoyproxy/controlplane/cache/SimpleCache.java

```diff
@@ -446,7 +445,7 @@ private Response createResponse(XdsRequest request, Map<String, VersionedResourc
         .map(VersionedResource::resource)
         .collect(Collectors.toList())
         : request.getResourceNamesList().stream()
-        .map(resources::get)
+        .map(name -> resources.getOrDefault(name, defaultResource(name, request.getResourceType())))
```

Comment:
safe to just return default here since we have checks before on missing resources.  Do not want to do checks again in private methods


```
if (!missingNames.isEmpty()) {
        if (watch.allowIncompleteEdsUpdate() && watch.request().getResourceType().equals(ResourceType.ENDPOINT)) {
          LOGGER.info(
              "responding incomplete EDS update in ADS mode for {} from node {} at version {} for request [{}] and [{}] not in snapshot",
              watch.request().getTypeUrl(),
              group,
              snapshot.version(watch.request().getResourceType(), watch.request().getResourceNamesList()),
              String.join(", ", watch.request().getResourceNamesList()),
              String.join(", ", missingNames));

        } else {
          LOGGER.info(
              "not responding in ADS mode for {} from node {} at version {} for request [{}] since [{}] not in snapshot",
              watch.request().getTypeUrl(),
              group,
              snapshot.version(watch.request().getResourceType(), watch.request().getResourceNamesList()),
              String.join(", ", watch.request().getResourceNamesList()),
              String.join(", ", missingNames));

          return false;
        }
    ```